dicesGame/src/test/java/com/example/dices/controllers/DiceControllerIntegrationTest.java

```diff
@@ -0,0 +1,266 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class DiceControllerIntegrationTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private IDiceRepository diceRepository;
+
+    @Autowired
+    private DiceService diceService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public static class TestConstants {
+        public static final int DEFAULT_DICE_ID1 = 1;
+        public static final int DEFAULT_DICE_SIZE1 = 6;
+        public static final int DEFAULT_DICE_ID2 = 5;
+        public static final int DEFAULT_DICE_SIZE2 = 9;
+        public static final int DEFAULT_DICE_NOT_EXIST = 5;
+        public static final int DEFAULT_DICE_INVALID_SIZE = 1001;
+        public static final String ERROR_MESSAGE_1 = "There is no information to display";
+        public static final String ERROR_MESSAGE_2 = "Dice not found with ID: ";
+        public static final String ERROR_MESSAGE_3 = "The dice size must be between 1 and 1000.";
+
+    }
+
+    @BeforeEach
+    public void setUp() {
+        // Initialize Databases
+
+        DiceModel diceModel = new DiceModel();
+        diceModel.setDiceId(TestConstants.DEFAULT_DICE_ID1);
+        diceModel.setDiceSize(TestConstants.DEFAULT_DICE_SIZE1);
+        diceModel.setDiceId(TestConstants.DEFAULT_DICE_ID2);
+        diceModel.setDiceSize(TestConstants.DEFAULT_DICE_SIZE2);
+        diceRepository.save(diceModel);
+    }
+
+    @AfterEach
+    public void cleanDatabase() {
+        // Clean repository after each test.
+        diceRepository.deleteAll();
+    }
+
+
+    @Test
+    void testGetDicesNoEmpty() throws Exception {
+        // makes the GET request to the /api/v1/dices endpoint
+        // Verify that the response has a status 200 (OK)
+        MvcResult result = mockMvc.perform(get("/api/v1/dices"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // Get the response as a JSON string and perform the necessary checks
+        String contentType = result.getResponse().getContentType();
+        assertEquals("application/json", contentType);
```

Comment:
Here you are only checking that the response is of type `application/json`, but you are not checking the actual response body.

Reply:
Added validation that the list is not empty.

Reply:
Where?